### PR TITLE
vexpress-qemu-flash: Use conditional assignments to allow overrides

### PR DESCRIPTION
**Pull request by drewmoseley**
Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

### CODE REVIEW

**Comment by drewmoseley** [comment]
@bboozzoo @kacf does this seem like a reasonable change?

File: meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf [hunk 1]
```diff
@@ -26,29 +26,29 @@ UBINIZE_ARGS = "-p 256KiB -m 1 -O 64"
 INSANE_SKIP_u-boot += "ldflags textrel"
 
 # combined NOR flash is 128MB
-MENDER_STORAGE_TOTAL_SIZE_MB = "128"
+MENDER_STORAGE_TOTAL_SIZE_MB ?= "128"
```
Comment (bboozzoo) [inline meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf]: don't think the storage will change

File: meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf [hunk 2]
```diff
@@ -13,8 +13,8 @@ IMAGE_FSTYPES = "ubifs vexpress-nor"
 MACHINE_ESSENTIAL_EXTRA_RDEPENDS = "kernel-image kernel-devicetree"
 MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += "mtd-utils mtd-utils-ubifs mtd-utils-jffs2"
 
-MKUBIFS_ARGS = "-m 1 -e 262016 -c 1024"
-UBINIZE_ARGS = "-p 256KiB -m 1 -O 64"
+MKUBIFS_ARGS ?= "-m 1 -e 262016 -c 1024"
```
Comment (bboozzoo) [inline meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf]: `MKUBIFS_ARGS` and `UBINIZE_ARGS` are probably not something you will need/want to change

File: meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf [hunk 3]
```diff
@@ -59,7 +59,7 @@ IMAGE_ROOTFS_MAXSIZE = "81920"
 # also used for generating fw_env.config but because of assumptions made in
 # meta-mender-core/recipes-bsp/u-boot/u-boot-fw-utils-mender.inc we need ship
 # our own config instead.
-MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET = "0x100000"
+MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET ?= "0x100000"
```
Comment (bboozzoo) [inline meta-mender-qemu/conf/machine/vexpress-qemu-flash.conf]: Changing this from local.conf will probably make u-boot out of sync with fw_env.config

**Comment by drewmoseley** [comment]
@bboozzoo thanks.  I updated the patch and removed the conditional assignment for the variables you mentioned.

**Comment by drewmoseley** [comment]
@bboozzoo any idea why the acceptance test failed?

**Comment by bboozzoo** [comment]
```
=================================== FAILURES ===================================
___________________________ TestCommits.test_commits ___________________________

self = <test_commits.TestCommits instance at 0x7f85cdb55560>

    def test_commits(self):
        # First find which range to check. Include HEAD and exclude all known
        # upstream branches.
        git_branch = subprocess.check_output(["git", "branch", "-r"])
        all_branches = [line.split()[0] for line in git_branch.strip().split('\n')]
    
        # Exclude all non-pull requests.
        commit_range = ["HEAD", "--not"]
        for branch in all_branches:
            # Include all non-origin branches.
            if not branch.startswith("origin/"):
                continue
            # Include branches that have slashes after "origin/" (pull requests).
            if re.match("^origin/.*/", branch):
                continue
            # Include branches that end with "patch-1" style text.
            if re.match("^origin/.*patch-[0-9]+$", branch):
                continue
            # Exclude if no matches above.
            commit_range.append(branch)
    
>       subprocess.check_call(["3rdparty/mendertesting/check_commits.sh"] + commit_range)
```
did you sign off the commit?

**Comment by drewmoseley** [comment]
Dang.  Forgot the changelog tag... again.  And I scrolled too quickly through the build output to see the log.  Thanks.

It's now fixed and as far as I can tell this is ready for merge.

**Comment by drewmoseley** [comment]
Hmm.  Now it seems to be a different error:
```
19:23:57 + go build
19:23:57 auth.go:7:2: cannot find package "github.com/pkg/errors" in any of:
19:23:57 	/usr/local/go/src/github.com/pkg/errors (from $GOROOT)
19:23:57 	/home/jenkins/workspace/yoctobuild/go/src/github.com/pkg/errors (from $GOPATH)
19:23:57 auth.go:8:2: cannot find package "github.com/sirupsen/logrus" in any of:
19:23:57 	/usr/local/go/src/github.com/sirupsen/logrus (from $GOROOT)
19:23:57 	/home/jenkins/workspace/yoctobuild/go/src/github.com/sirupsen/logrus (from $GOPATH)
19:23:57 + testFinished
19:23:57 + [[ 0 -eq 0 ]]
19:23:57 + github_pull_request_comment 'Jenkins build [job](https://mender-jenkins.mender.io/job/mender-builder/1721/) failed.'
19:23:57 ++ cat
19:23:57 + local 'request_body=    {
19:23:57       "body": "Jenkins build [job](https://mender-jenkins.mender.io/job/mender-builder/1721/) failed."
19:23:57     }'
19:23:57 + curl --user mender-test-bot:bWVuZGVyCg1 -d '    {
19:23:57       "body": "Jenkins build [job](https://mender-jenkins.mender.io/job/mender-builder/1721/) failed."
```

**Comment by bboozzoo** [comment]
@GregorioDiStefano error that @drewmoseley, is it related to mender-stress-client? I thought this was already fixed in master.

**Comment by kacf** [comment]
Can you rebase it on master to trigger a new build? I think the failure should be fixed now.

**Comment by drewmoseley** [comment]
@bboozzoo I'm not sure what mender-stress-client is.  I'll do the rebase and hopefully that's all we need.  Let me know if you have additional concerns.

**Comment by drewmoseley** [comment]
Rebased and force pushed to update my branch.  Hopefully the automated test will succeed now.

**Comment by bboozzoo** [comment]
```Parsing recipes...ERROR: ExpansionError during parsing /home/jenkins/workspace/yoctobuild/meta-mender/meta-mender-core/recipes-mender/mender/mender_git.bb
Traceback (most recent call last):
  File "/home/jenkins/workspace/yoctobuild/bitbake/lib/bb/data_smart.py", line 412, in DataSmart.expandWithRefs(s='${MENDER_BRANCH}-git${SRCPV}', varname='PV'):
                 try:
    >                s = __expand_var_regexp__.sub(varparse.var_sub, s)
                     try:
  File "/home/jenkins/workspace/yoctobuild/bitbake/lib/bb/data_smart.py", line 111, in VariableParse.var_sub(match=<_sre.SRE_Match object; span=(0, 16), match='${MENDER_BRANCH}'>):
                 else:
    >                var = self.d.getVarFlag(key, "_content")
                 self.references.add(key)
  File "/home/jenkins/workspace/yoctobuild/bitbake/lib/bb/data_smart.py", line 794, in DataSmart.getVarFlag(var='MENDER_BRANCH', flag='_content', expand=True, noweakdefault=False, parsing=False):
                     cachename = var + "[" + flag + "]"
    >            value = self.expand(value, cachename)
     
  File "/home/jenkins/workspace/yoctobuild/bitbake/lib/bb/data_smart.py", line 436, in DataSmart.expand(s="${@mender_branch_from_preferred_version(d.getVar('PREFERRED_VERSION'))}", varname='MENDER_BRANCH'):
         def expand(self, s, varname = None):
    >        return self.expandWithRefs(s, varname).value
     
  File "/home/jenkins/workspace/yoctobuild/bitbake/lib/bb/data_smart.py", line 426, in DataSmart.expandWithRefs(s="${@mender_branch_from_preferred_version(d.getVar('PREFERRED_VERSION'))}", varname='MENDER_BRANCH'):
                 except Exception as exc:
    >                raise ExpansionError(varname, s, exc) from exc
     
bb.data_smart.ExpansionError: Failure expanding variable MENDER_BRANCH, expression was ${@mender_branch_from_preferred_version(d.getVar('PREFERRED_VERSION'))} which triggered exception ValueError: substring not found
```
hmm interesting, don't think it's related to your PR though

**Comment by kacf** [comment]
Can you try yet another rebase? :-)

**Comment by drewmoseley** [comment]
Done.  Hoping for a successful build.

**Comment by drewmoseley** [comment]
Hooray.  Looks like the build issues are resolved.

**Comment by drewmoseley** [comment]
@kacf is this ok to merge now?

**Comment by kacf** [comment]
Yes, sorry, this got left behind!

**Comment by drewmoseley** [comment]
No problem.  Thanks for the quick response.

**Pull request merged**